mne/event.py

```diff
@@ -885,3 +886,242 @@ def concatenate_events(events, first_samps, last_samps):
+                post_ok = (
+                    np.bitwise_and(ev['newbits'],
+                                   mne_events[:, 2]) == ev['newbits'])
+                ok_ind = np.where(pre_ok & post_ok)
+                events_[ok_ind, 2] |= 1 << (n - 1)
+        return events_
+
+    def _mne_events_to_category_t0(self, cat, mne_events, sfreq):
+        """ Translate mne_events to reference times (t0) for epochs in a given
+        DACQ averaging category cat. """
+        cat_ev = cat['event']
+        cat_reqev = cat['reqevent']
+        events = self._events_mne_to_dacq(mne_events)
+        times = events[:, 0]
+        # indices of times where ref. event occurs
+        refEvents_inds = np.where(events[:, 2] & (1 << cat_ev - 1))[0]
+        refEvents_t = times[refEvents_inds]
+        if cat_reqev:
+            # indices of times where req. event occurs
+            reqEvents_inds = np.where(events[:, 2] & (
+                1 << cat_reqev - 1))[0]
+            reqEvents_t = times[reqEvents_inds]
+            # relative (to refevent) time window where req. event
+            # must occur (e.g. [0 .2])
+            twin = [0, (-1)**(cat['reqwhen']) * cat['reqwithin']]
+            win = np.round(np.array(sorted(twin)) * sfreq)  # to samples
+            refEvents_wins = refEvents_t[:, None] + win
+            req_acc = np.zeros(refEvents_inds.shape, dtype=bool)
+            for t in reqEvents_t:
+                # mark time windows where req. condition is satisfied
+                reqEvent_in_win = np.logical_and(
+                    t >= refEvents_wins[:, 0], t <= refEvents_wins[:, 1])
+                req_acc |= reqEvent_in_win
+            # drop ref. events where req. event condition is not satisfied
+            refEvents_inds = refEvents_inds[np.where(req_acc)]
+            refEvents_t = times[refEvents_inds]
+        # adjust for trigger-stimulus delay by delaying the ref. event
+        refEvents_t += int(np.round(self._events[cat_ev]['delay'] * sfreq))
+        return refEvents_t
+
+    @property
+    def categories(self):
+        """ Return list of category names in DACQ defined order. Only returns
+        categories marked active in DACQ. """
+        cats = sorted(self._categories_in_use.values(),
+                      key=lambda cat: cat['index'])
+        return cats
+
+    @property
+    def events(self):
+        """ Lists events in DACQ defined order. Only returns events
+        that are in use (referred to by a category). """
+        evs = sorted(self._events_in_use.values(),
+                     key=lambda ev: ev['index'])
+        return evs
+
+    @property
+    def event_names(self):
+        return [event['comment'] for event in self.events]
+
+    def get_epochs(self, raw, category, picks=None, reject=None,
+                   baseline=(None, 0), stim_channel=None, mask=0):
+        """ Get mne.Epochs instance corresponding to the given category. """
```

Comment:
Indeed. A lot of the parameters are just passed to `mne.Epochs`, so I guess I'll copy the relevant parts of that docstring.